fix from upstream for a progress bar error in some locales

http://bugs.gentoo.org/208484

--- src/progress.c	2008-01-26 11:26:58.000000000 +0200
+++ src/progress.c	2008-02-11 10:12:42.000000000 +0200
@@ -969,8 +969,6 @@
       move_to_end (p);
     }
 
-  assert (p - bp->buffer <= bp->width);
-
   while (p < bp->buffer + bp->width)
     *p++ = ' ';
   *p = '\0';
